# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PreallocatedProcessManager.cpp
# Commit: fc2642d29858
# Full Hash: fc2642d2985823c1ae64e8d302de042344920e79
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-07-24 09:50:25
# Description:
#   Bug 1908789 - Check for null return from MakePreallocProcess, r=ipc-reviewers,mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217103
# ==============================================================================

diff -r 25faf44f8965 -r fc2642d29858 dom/ipc/PreallocatedProcessManager.cpp
--- a/dom/ipc/PreallocatedProcessManager.cpp	Tue Jul 23 20:17:51 2024 +0000
+++ b/dom/ipc/PreallocatedProcessManager.cpp	Tue Jul 23 20:17:52 2024 +0000
@@ -316,6 +316,14 @@
   }
 
   UniqueContentParentKeepAlive process = ContentParent::MakePreallocProcess();
+  if (!process) {
+    // We fully failed to create a prealloc process. Don't try to kick off a new
+    // preallocation here, to avoid possible looping.
+    MOZ_LOG(ContentParent::GetLog(), LogLevel::Debug,
+            ("Failed to launch prealloc process"));
+    return;
+  }
+
   process->WaitForLaunchAsync(PROCESS_PRIORITY_PREALLOC)
       ->Then(
           GetCurrentSerialEventTarget(), __func__,
